qds_sdk/commands.py

```diff
@@ -87,10 +87,26 @@ def run(cls, **kwargs):
         Returns:
             Command object
         """
+        err_pointer, tmp_pointer, new_bytes = 0, 0, 0
         cmd = cls.create(**kwargs)
         while not Command.is_done(cmd.status):
             time.sleep(Qubole.poll_interval)
             cmd = cls.find(cmd.id)
+            if kwargs.get('print_logs_live', False):
+                log, err_length, tmp_length = cmd.get_log_partial(err_pointer, tmp_pointer)
+
+                if err_length != "0":
+                    err_pointer += int(err_length)
+                    new_bytes = int(err_length) + int(tmp_length) - new_bytes - tmp_pointer
+                    tmp_pointer = int(tmp_length)
+
+                else:
+                    tmp_pointer += int(tmp_length)
+                    new_bytes = int(tmp_length)
+
+                if len(log) > 0 and new_bytes > 0:
+                    data = list(log)
+                    print >>sys.stderr, "".join(data[-new_bytes:])
```

Comment:
I guess we don't need to do list operation on the log variable and then do join operation. we can simply do log[-new_bytes:].